Library/Homebrew/test/Rakefile

```diff
@@ -1,8 +1,14 @@
 require "rake"
 require "rake/testtask"
 
+def mac?
+  /darwin/i === RUBY_PLATFORM && !ENV["HOMEBREW_TEST_GENERIC_OS"]
+end
+
 TEST_DIRECTORY = File.dirname(File.expand_path(__FILE__))
-TEST_FILES = Dir["#{TEST_DIRECTORY}/test_*.rb"]
+TEST_FILES = Dir["#{TEST_DIRECTORY}/test_*.rb"].reject do |f|
+  f.include?("/test_os_mac_") && !mac?
+end
```

Comment:
I don't like the "skipping" here. We should endeavour to push platform-specific code into different files (but I'm definitely open to a different approach within those constraints).